### DISCUSSION TITLE
NN-UNet

### DISCUSSION THREAD

**Discussion by iffthomas** [Project-MONAI/MONAI#7191] [Q&A]
Hi, Thank you for implementing an nnUnet api for monai.

I'm currently playing around with your implementation.
I'm curious if there exist a possiblity to change the trainer_class_name to nnUNetTrainer_250epoch while only training for the 2D case on all folds I have (I've got 5).
what would be a suitable approach to achieve this?
I've tried adding --config "2d" that my command would look like this
python -m monai.apps.nnunet nnUNetV2Runner run --input_config='./input.yaml' --trainer_class_name nnUNetTrainer_1epoch --config "2d"

**Comment by KumoLiu** [comment]
Hi @iffthomas, for how to run nnunet, you can refer to this [tutorial](https://github.com/Project-MONAI/tutorials/tree/main/nnunet).
https://github.com/Project-MONAI/MONAI/blob/2658b00a88cb6751ac155cc2d133e35aba0d30de/monai/apps/nnunet/nnunetv2_runner.py#L957
https://github.com/Project-MONAI/MONAI/blob/2658b00a88cb6751ac155cc2d133e35aba0d30de/monai/apps/nnunet/nnunetv2_runner.py#L559
If you use the run command, it will run all the configs by default.

Hope it helps, thanks!

**Reply by iffthomas** [reply]
Hi @KumoLiu 

Thank you very much!
I didn't know that I can input additional args to the run command such as the needed --configs "2d"

**Reply by iffthomas** [reply]
Also What I figured out, 
python -m monai.apps.nnunet nnUNetV2Runner train --input_config='./input.yaml' --trainer_class_name nnUNetTrainer_250epoch --config "2d" get me an error message, is this intended?


below my logs
    nnunet_trainer = get_trainer_from_args(dataset_name_or_id, configuration, fold, trainer_class_name,
  File "/cluster/home/ifft/miniconda3/envs/venv-spine/lib/python3.10/site-packages/nnunetv2/run/run_training.py", line 42, in get_trainer_from_args
    raise RuntimeError(f'Could not find requested nnunet trainer {trainer_name} in '
RuntimeError: Could not find requested nnunet trainer nnUNetTrainer_250epoch in nnunetv2.training.nnUNetTrainer (/cluster/home/ifft/miniconda3/envs/venv-spine/lib/python3.10/site-packages/nnunetv2/training/nnUNetTrainer). If it is located somewhere else, please move it there.

**Reply by KumoLiu** [reply]
Hi @iffthomas, 
- If you want to use the one-line command `run` to start the nnU-Net workflow, you can modify the [train](https://github.com/Project-MONAI/MONAI/blob/2658b00a88cb6751ac155cc2d133e35aba0d30de/monai/apps/nnunet/nnunetv2_runner.py#L957) here manually by specify the "configs='2d'". 
- And for the error, thanks for pointing out the typo [here](https://github.com/Project-MONAI/MONAI/issues/7192#issuecomment-1792233134), I will create a PR to fix it.
Thanks!

**Reply by iffthomas** [reply]
Sorry, I'm rather new to programming in general,

- If you want to use the one-line command run to start the nnU-Net workflow, you can modify the [train](https://github.com/Project-MONAI/MONAI/blob/2658b00a88cb6751ac155cc2d133e35aba0d30de/monai/apps/nnunet/nnunetv2_runner.py#L957) here manually by specify the "configs='2d'".

what the best approach in doing this?
Go to my downloaded files where I've installed Monai NN-Unet and then changeing it there?

**Reply by KumoLiu** [reply]
Yes, but it's not always a good habit to modify the source code, so instead of using the one-line command, you could run it separately, step by step, so that you can specify the config to be used during the train. 
https://github.com/Project-MONAI/MONAI/blob/fbbb64b4cb2facffccbf4fdf6c44e20f8d31ca10/monai/apps/nnunet/nnunetv2_runner.py#L117

Perhaps we could add more flexibility to this one-line command? What do you think? cc @dongyang0122 
https://github.com/Project-MONAI/MONAI/blob/fbbb64b4cb2facffccbf4fdf6c44e20f8d31ca10/monai/apps/nnunet/nnunetv2_runner.py#L932

**Reply by iffthomas** [reply]
Ah yes, currently I'm doing this with just running the train stage with this command prior to plan & preprocessing,
Thank you very much for taking your time answering my questions
